crates/goose-mcp/src/developer/mod.rs

```diff
@@ -1715,17 +1794,57 @@ mod tests {
         let dir = TempDir::new().unwrap();
         std::env::set_current_dir(dir.path()).unwrap();
 
-        let router = DeveloperRouter::new();
+        // Make sure there are no global hints
```

Comment:
I know this is the existing pattern here, but it is not ideal; if the test fails half way, you lose your .goosehints file 

something like:

```
struct FileBackup {
    path: PathBuf,
    original_content: Option<String>,
}

impl Drop for FileBackup {
    fn drop(&mut self) {
        if let Some(content) = &self.original_content {
            let _ = fs::write(&self.path, content);
        }
    }
}
```

would be a better solution